File: BaekJoon/Silver/11723/C++/11723.cpp
// no.11723: 집합 (S5)

#include <cstdio>
#include <cstring>
using namespace std;

int main() {
    int m;
    scanf("%d", &m);
    int s=0;
    for(int i=0; i<m; i++) {
        char cmd[10];
        scanf("%s", cmd);
        if(strcmp(cmd, "add")==0) {
            int x;
            scanf("%d", &x);
            s = s|(1<<x);
        }
        if(strcmp(cmd, "remove")==0) {
            int x;
            scanf("%d", &x);
            s = s&~(1<<x);
        }
        if(strcmp(cmd, "check")==0) {
            int x;
            scanf("%d", &x);
            if(s&(1<<x))
                printf("1\n");
            else
                printf("0\n");
        }
        if(strcmp(cmd, "toggle")==0) {
            int x;
            scanf("%d", &x);
            if(s&(1<<x))
                s = s&~(1<<x);
            else
                s = s|(1<<x);
        }
        if(strcmp(cmd, "all")==0) {
            s = ~s;
        }
        if(strcmp(cmd, "empty")==0) {
            s = 0;
        }
    }
    return 0;
}

File: BaekJoon/Silver/11723/Python/11723.py
import sys

m = int(sys.stdin.readline())
set1 = set()

for i in range(m):
    cmd = sys.stdin.readline().rstrip()
    if 'add' in cmd:
        n = int(cmd.split().pop())
        if n not in set1:
            set1.add(n)
    elif 'remove' in cmd:
        n = int(cmd.split().pop())
        if n in set1:
            set1.remove(n)
    elif 'check' in cmd:
        n = int(cmd.split().pop())
        print(1 if n in set1 else 0)
    elif 'toggle' in cmd:
        n = int(cmd.split().pop())
        if n in set1:
            set1.remove(n)
        else:
            set1.add(n)
    elif cmd == 'all':
        set1 = set(range(1, 21))
    else:
        set1.clear()